tests/jest/plugins/search/plugin.search.test.js

```diff
@@ -123,28 +123,26 @@ describe('Plugin: Search', () => {
     expect(br.options.goToFirstResult).toBeTruthy();
   });
 
-  test('SearchCallback event fires when AJAX search returns results', () => {
+  test('SearchCallback event fires when AJAX search returns results', async () => {
     br.init();
     const dfd = br.search('foo');
-    return dfd.then(() => {
-      expect(triggeredEvents()).toContain(`${namespace}SearchCallback`);
-    });
+    await expect(triggeredEvents()).toContain(`${namespace}SearchCallback`);
```

Comment:
`br.search` returns a promise (that's why we call `.then` on its return); so that's the thing we want to wait for. Actually we can probably remove `dfd` entirely and just call `await br.search('foo')`.

Similarly for the other cases in the file.